src/common/types/timestamp.cpp

```diff
@@ -298,44 +298,62 @@ timestamp_t Timestamp::GetCurrentTimestamp() {
 
 timestamp_t Timestamp::FromEpochSeconds(int64_t sec) {
 	int64_t result;
+	D_ASSERT(Timestamp::IsFinite(timestamp_t(sec)));
 	if (!TryMultiplyOperator::Operation(sec, Interval::MICROS_PER_SEC, result)) {
 		throw ConversionException("Could not convert Timestamp(S) to Timestamp(US)");
 	}
 	return timestamp_t(result);
 }
 
-timestamp_t Timestamp::FromEpochMs(int64_t ms) {
+timestamp_t Timestamp::FromEpochMsPossiblyInfinite(int64_t ms) {
 	int64_t result;
 	if (!TryMultiplyOperator::Operation(ms, Interval::MICROS_PER_MSEC, result)) {
 		throw ConversionException("Could not convert Timestamp(MS) to Timestamp(US)");
 	}
 	return timestamp_t(result);
 }
 
+timestamp_t Timestamp::FromEpochMs(int64_t ms) {
+	D_ASSERT(Timestamp::IsFinite(timestamp_t(ms)));
+	return FromEpochMsPossiblyInfinite(ms);
+}
+
 timestamp_t Timestamp::FromEpochMicroSeconds(int64_t micros) {
 	return timestamp_t(micros);
 }
 
 timestamp_t Timestamp::FromEpochNanoSeconds(int64_t ns) {
+	D_ASSERT(Timestamp::IsFinite(timestamp_t(ns)));
 	return timestamp_t(ns / 1000);
```

Comment:
There is an Interval constant for this.

Reply:
Not your fault but may fix it now?

Reply:
I'm not following, we receive a timestamp and output a timestamp, how is this related to interval?